File: app/src/main/java/com/theappsolutions/boilerplate/other/functions/Function2.java
package com.theappsolutions.boilerplate.other.functions;

/**
 * @author Severyn Parkhomenko <sparkhomenko@grossum.com>
 * @copyright (c) Grossum. (http://www.grossum.com)
 */
public interface Function2<T1, T2, R> {
    R call(T1 param1, T2 param2);
}

File: app/src/main/java/com/theappsolutions/boilerplate/other/functions/Function2.kt
package com.theappsolutions.boilerplate.other.functions

/**
 * @author Severyn Parkhomenko s.parkhomenko@theappsolutions.com
 * @copyright (c) 2018 TheAppSolutions. (https://theappsolutions.com)
 */
interface Function2<T1, T2, R> {
    fun call(param1: T1, param2: T2): R
}
